include/spinlock.h

```diff
@@ -33,7 +33,7 @@ static inline void z_arch_irq_unlock(int key)
  * even more poorly supported.
  */
 #if (CONFIG_FLASH_SIZE == 0) || (CONFIG_FLASH_SIZE > 32)
-#if defined(CONFIG_ASSERT) && (CONFIG_MP_NUM_CPUS < 4)
+#if defined(CONFIG_ASSERT) && (CONFIG_MP_NUM_CPUS < 2)
```

Comment:
No... that's wrong.  This disables spinlock validation whenever we're running with more than one CPU.  Why is this change here?  The test for CPU count is because I'm (somewhat sneakily) packing the CPU index into the bottom two bits of the current thread pointer for testing ownership of the spinlock, and there are only two such bits available.

I know @dcpleung reported some problems with validation in SMP on hardware too (though oddly I don't see them on x86_64), so if you need to disable SPIN_VALIDATE on SMP that's probably OK.  Just do it in a separate test and not in this line.

Reply:
@andyross  thanks for the review. 
While I'm doing the debug for this case, I found when two different threads running on different cpu try to access the same one spinlock or semaphore concurrently, there will be assert. The situation is that one thread get the lock(set the owner) and didn't do the unlock, the other thread exactly do the spin validate.

OK, I will do the disable in another PR.